docs/first_semantic_game/02.simple_chat_npcs.rst

```diff
@@ -1,163 +1,395 @@
+can hold interactive conversations.  It assumes that you have completed the 
+project setup and have configured the OPENAI_API_KEY user secret.
 
-With the Semantic Kernel set up and Godot Project running, the next step is to create a basic NPC for the player to interact with. Here we will use simple shapes for NPCs instead of more lifelike characters.
+The NPC Sprite
+--------------
 
-Step 1: Creating the NPC Sprite
-**********************************
+Here we will create a basic sprite that we can attach chat functionality to.
 
-- Switch to the 2D workspace in Godot.
-- In the Scene tab, click the "+" symbol to add a 'Node'. In the "Create New Node" window, Double click on `Area2D` to add it to the scene. This will serve as the basis of your NPC.
-- Rename the `Area2D` node to 'NPC' or any name you prefer.
-- With the 'NPC' node selected, add another new node under it. This time, choose a `CollisionShape2D`. This is important as it dictates the clickable interactable area for your NPC.
-- To create a visible entity for the NPC, create another child node under the `CollisionShape2D` called `Sprite`. Here you can upload or select any simple 2D shape image you have available.
+1. In the ``MainGame`` scene that we created in :doc:`01.project_setup`
+   insert an ``Area2D`` node and rename it to  ``NPC``.
+2. Add a ``CollisionShape2D`` rectangle to the NPC. This will be useful 
+   later and removes the yellow warning in the workspace.
+3. With the ``NPC`` node selected, add a ``Sprite2D`` node. This 
+   makes a visible sprite for our character.
+4. To set the image for our NPC sprite, copy the troll portrait into 
+   the project. Then locate and choose the image as an ``ImageTexture`` 
+   in the properties panel for the ``Sprite2D`` node.
+5. Drag the NPC image near the center of the scene. Build and run the project to
+   see the river troll NPC token.
 
-Step 2: Setting up the NPC Node Architecture
-*************************************************
+Meet Gralk:
 
-Right click on the 'NPC' node and select "Attach Script". Pick an intuitive name for the script and hit "Create".
+.. image:: gralk.png
+   :alt: The Gralk NPC sprite
+   :width: 150px
+   :align: center
 
-Player Character Creation and Movement
-----------------------------------------
 
-TODO
+The ChatEntity class
+--------------------
+
+Here we a define a base class called ``ChatEntity`` for all chat-enabled  entities in the game. 
+This type encapsulates basic properties like a name and a description, and, later, 
+it handles behaviors such as detecting other nearby chat entities. Subsequently, 
+we will extend this node type to create ``ChatAI`` for GPT-4 driven NPCs and 
+``ChatPlayer`` for player characters.
+
+Right-click the filesystem dock and select "New Script...". **Make sure the language 
+is set to C#** and it inherits from ``Area2D``.
+
+Save this script as ``ChatEntity.cs`` in the project's root directory.
+

```

Comment:
Should we insert the instructions about adding the `[GlobalClass]` attribute here?

Reply:
```suggestion

To register this script with the Godot editor, add the ``[GlobalClass]`` attribute right above the class declaration.

.. code-block:: csharp

    [GlobalClass]
    public partial class ChatEntity : Area2D
    {
        ...
    }

For more information about Godot's global classes, see https://docs.godotengine.org/en/stable/tutorials/scripting/c_sharp/c_sharp_global_classes.html

```